ang/test/mocks/data/cases-types.data.js

```diff
@@ -408,7 +411,16 @@
      * @param {object} newCaseTypes a map of case types indexed by their id.
      */
     this.add = function (newCaseTypes) {
-      caseTypesMock = _.assign({}, caseTypesMock, newCaseTypes);
+      var newCaseTypesMock = _.assign({}, caseTypesMock, newCaseTypes);
+
+      CRM['civicase-base'].caseTypes = _.clone(newCaseTypesMock);
+    };
+
+    /**
+     * Restores the mock case types to the default ones. This avoids different
+     * spec files from overriding values to other files.
+     */
+    this.restore = function () {
```

Comment:
But we are already using `clone` to avoid this problem. Not sure why this is required?

Reply:
@deb1990 this implementation was wrong, I was just trying to use the code we previously had. Cleaned it and made sure it works. 

It does need to be restored even if we use `clone` because we need to update the `CRM['civicase-base'].caseTypes` object. These values are going to be carried to other tests. I added a console log after adding a case type and removed the reset statement and this is how it looks:
<img width="1992" alt="Screen Shot 2021-01-06 at 9 24 54 PM" src="https://user-images.githubusercontent.com/1642119/103844357-18f1ef80-5070-11eb-9660-5037c951a9d3.png">

There was a similar function already, but was not ready at provision level so I'm using the same one. That was part of the cleanup.